dowhy/causal_refuters/bootstrap_refuter.py

```diff
@@ -1,33 +1,75 @@
     """
-    Refute an estimate by running it on a random sample of the original data.
+    Refute an estimate by running it on a random sample of the data containing measurement error in the 
+    confounders. This allows us to find the ability of the estimator to find the effect of the 
+    treatment on the outcome.
+    
     It supports additional parameters that can be specified in the refute_estimate() method.
-    - 'num_simulations': int, CausalRefuter.DEFAULT_NUM_SIMULATIONS by default
-    The number of bootstrap simulations to be run
+    
+    Parameters
+    -----------
+    -'num_simulations': int, CausalRefuter.DEFAULT_NUM_SIMULATIONS by default
+    The number of simulations to be run
     - 'sample_size': int, Size of the original data by default
     The size of each bootstrap sample
+    - 'required_variables': int, list
+    An integer argument means that we select a select number of covariates out of all covariates
+    - 'noise': float, BootstrapRefuter.DEFAULT_STD_DEV by default
+    The standard deviation of the noise to be added to the data
     - 'random_state': int, RandomState, None by default
     The seed value to be added if we wish to repeat the same random behavior. For this purpose, 
     we repeat the same seed in the psuedo-random generator.
     """
 
+    DEFAULT_STD_DEV = 0.1
+    DEFAULT_SUCCESS_PROBABILITY = 0.5
+    DEFAULT_NUMBER_OF_TRIALS = 1
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self._num_simulations = kwargs.pop("num_simulations", CausalRefuter.DEFAULT_NUM_SIMULATIONS)
-        self._sample_size = kwargs.pop("sample_size",len(self._data))
-        self._random_state = kwargs.pop("random_state",None)
+        self._num_simulations = kwargs.pop("num_simulations", CausalRefuter.DEFAULT_NUM_SIMULATIONS )
+        self._sample_size = kwargs.pop("sample_size", len(self._data))
+        self._required_variables = kwargs.pop("required_variables", None)
+        self._noise = kwargs.pop("noise", BootstrapRefuter.DEFAULT_STD_DEV )
+        self._random_state = kwargs.pop("random_state", None)
 
         if 'logging_level' in kwargs:
             logging.basicConfig(level=kwargs['logging_level'])
         else:
             logging.basicConfig(level=logging.INFO)
         self.logger = logging.getLogger(__name__)
         
+        # Sanity check the data passed by the user
+        # If the data is invalid, we run the default behavior
+        if self._required_variables is int:
+            if len(self._target_estimand.backdoor_variables) < self._required_variables:
+                self.logger.warning("Too many variables passed.\n The number of backdoor variables is: {}.\n The number of variables passed: {}".format(
+                    len(self._target_estimand.backdoor_variables),
+                    self._required_variables )
+                )
+                self.logger.warning("The bootstrap refuter will follow the default behavior")

```

Comment:
I think it's better to throw logger warning and a ValueError exception here. Automatic defaults can be confusing for a user if they ignore the warning message.